pkg/cache/cache_test.go

```diff
@@ -1822,12 +1810,11 @@ func isKubeService(svc metav1.Object) bool {
 }
 
 func isPodDisableDeepCopy(opts cache.Options) bool {
-	if d, ok := opts.View.ByObject.UnsafeDisableDeepCopy[&corev1.Pod{}]; ok {
-		return d
-	} else if d, ok = opts.View.ByObject.UnsafeDisableDeepCopy[cache.ObjectAll{}]; ok {
-		return d
-	} else if d, ok = opts.View.ByObject.UnsafeDisableDeepCopy[&cache.ObjectAll{}]; ok {
-		return d
+	if opts.ByObject[&corev1.Pod{}].UnsafeDisableDeepCopy != nil && *opts.ByObject[&corev1.Pod{}].UnsafeDisableDeepCopy {
+		return true
```

Comment:
Shouldn't this return the value of UnsafeDisableDeepCopy?

(same below)

Reply:
I.e. if it's set to false on the object shouldn't it be false?

Reply:
Changed it, there shouldn't be any logical difference, but I agree it might be more clear

Reply:
Actually I stand corrected, we want to always return the most specific value now, given that's a pointer

Reply:
Yup, that's what I thought :)